spec/features/items/index_html.rb

```diff
@@ -0,0 +1,37 @@
+require "rails_helper"
+
+RSpec.describe "Items Show" do 
+  describe "as a visitor" do 
+    before(:each) do
+      @supermarket_1 = Supermarket.create!(name: "Sprouts", location: "CO")
+
+      @item_1 = @supermarket_1.items.create!(name: "Beef", price: 5)
+      @item_2 = @supermarket_1.items.create!(name: "Cheese", price: 3)
+      @item_3 = @supermarket_1.items.create!(name: "Soup", price: 2)
+
+      @customer_1 = Customer.create!(name: "Aaron")
+      @customer_2 = Customer.create!(name: "Chase")
+
+      CustomerItems.create!(customer: @customer_1, item: @item_1)
+      CustomerItems.create!(customer: @customer_1, item: @item_2)
+      CustomerItems.create!(customer: @customer_2, item: @item_2)
+      CustomerItems.create!(customer: @customer_2, item: @item_3)
+    end
+
+    # User Story 3
+    it "displays a list of all items & attributes" do 
+      visit "/items"
+
+      expect(page).to have_content("Name: #{@item_1.name}")
+      expect(page).to have_content("Price: #{@item_1.price}")
+      expect(page).to have_content("From: #{@item_1.supermarket.name}")
+    end
+
+    # User Story 3
+    it "displays the count of customers that bought the item" do
+      visit "/items"
+
+      expect(page).to have_content("Count: #{@item_1.customer_count}")
```

Comment:
The structure of this test is excellent -- my only suggestion would be to not call the same `customer_count` method that you are testing. 

Consider the possibility that your logic in the `customer_count` method is somehow incorrect (it isn't, but let's imagine :) ). If you were to call this same method in the test, then your test would pass because it would be calling this incorrect method in order to verify that the method works. So, a more meaningful assertion is to do the calculation manually (it's pretty easy in this case - Item 1 is only associated with 1 customer) and use that result in the assertion directly: `expect(page).to have_content("Count: 1")`

That way, we can be sure that the functionality of the method is behaving correctly.

Reply:
Since this is such a simple case bc there is only one customer, this is another reason why we always want to test these model methods in a model spec as well. It's a bit easier to set up data there so we can test more complex use cases (like when there is more than just one customer associated to an item)